### PR TITLE
add prelogin_url template tag to generate prefix-aware url

### PR DESCRIPTION
**Pull request by sravfeyn**

### CODE REVIEW

**Comment by orangejenny** [comment]
Test failure looks real:

TemplateSyntaxError: Invalid block tag: 'prelogin_url'

**Comment by sravfeyn** [comment]
This is weird, I am not able to reproduce on local machine. And I did create `prelogin_url` template tag correctly

File: corehq/apps/hqwebapp/templatetags/menu_tags.py [hunk 1]
```diff
@@ -159,3 +160,14 @@ def aliased_language_name(lang_code):
...
         raise KeyError('Unknown language code %s' % lang_code)
+
+
+@register.simple_tag(takes_context=True)
+def prelogin_url(context, urlname):
+    """
+    A prefix aware url tag replacement for prelogin URLs
+    """
+    if context.get('url_uses_prefix', False):
+        return reverse(urlname, args=[context['LANGUAGE_CODE']])
```
Comment (czue) [inline corehq/apps/hqwebapp/templatetags/menu_tags.py]: remind me where `LANGUAGE_CODE` gets in the context? should we consider also checking if it's present so this doesn't crash if it's not?

Comment (sravfeyn) [inline corehq/apps/hqwebapp/templatetags/menu_tags.py]: It comes from [Django i18n context processor](https://docs.djangoproject.com/en/1.9/ref/templates/api/#django-template-context-processors-i18n)

Unless a view doesn't use (all prelogin view use) the `RequestContext` `LANGUAGE_CODE` should always be available. Will add a check and default to non-prefix URL

Comment (sravfeyn) [inline corehq/apps/hqwebapp/templatetags/menu_tags.py]: @czue updated

**Pull request merged**